### ISSUE TITLE
Toy example with synthetic data

### ISSUE THREAD

**Issue by aymuos15** [JDSobek/MedYOLO#42]
Do you have the bandwidth to create a toy example which overfits on some synthetic data?

**Comment by JDSobek** [comment]
Probably not, especially considering the required compute, but if you can get the [BRATS 2021 Task 1](https://www.kaggle.com/datasets/dschettler8845/brats-2021-task1) dataset [this is the code I used to convert the BRATS segmentations into MedYOLO labels.](https://github.com/JDSobek/PublicMLCode/blob/main/misc/BRATS_tools.py)

I haven't looked at the code for awhile so you should look through it before running it, but it should be close to what you need to get a dataset ready for trial runs. You might just be able to set the input and output folders and run it to get the T1CE dataset ready.

**Comment by aymuos15** [comment]
Sweet! Thanks a ton!

One more question. What do you think is the most minimal gpu resource I can set up (assuming I do not care for performance, just testing).

It is possible to do: MedYOLO-S 2 350 25 (from table 1 of the paper).

Also, can we not do batch size 1?

**Comment by JDSobek** [comment]
Since the paper was published I added a [nano config file](https://github.com/JDSobek/MedYOLO/blob/main/models3D/yolo3Dn.yaml) version after a few requests, which emulates the smallest YOLOv5 model. I haven't tested it though, and I don't remember any reports on its performance, so I can't attest to how well it works or how much VRAM it consumes.

I wouldn't go smaller than 350 for the edge length, I think that's about as small as I could go before I started running into issues from the downsampling.

I don't remember if I ever tried batch size 1, and if I did I don't remember what happened. I suppose there shouldn't be any harm in trying it, I might have avoided it just to preserve more batch randomization during weight updates.

I remember seeing the BRATS training runs diverge sometimes, but other than it consistently happening with the largest model I never figured out why. If it happens to you it might just need to be re-run and hope for better initialization, or it might need some hyperparameter adjustment.

**Comment by aymuos15** [comment]
Awesome.

Thank you so much. Really appreciate the detailed response.
